### DISCUSSION TITLE
OTel Tracing for Pydantic Model Creations and Validations

### DISCUSSION THREAD

**Discussion by cyberksh** [pydantic/pydantic#9666] [General]
Is there an OTel plugin that I can use to enable tracing for pydantic model generation in my code? I'm trying to improve tracing capabilities to my APIs.

**Comment by samuelcolvin** [comment]
Yes! We spent the last year building it.

[Pydantic Logfire](https://pydantic.dev/logfire), specifically the [pydantic plugin](https://docs.pydantic.dev/logfire/integrations/pydantic/).
